### ISSUE TITLE
Test with Boost 1.63

### ISSUE THREAD

**Issue by guwirth** [etas/vs-boost-unit-test-adapter#168]
BUTA should be compatible with Boost 1.63. Please start testing before release of 1.63 to give feedback in case of problems.

**Comment by netspiri** [comment]
Boost 1.63-beta is to be released on
Wednesday, November 16, 2016 according to http://www.boost.org/development/index.html.

**Comment by guwirth** [comment]
@netspiri please test it. This time should work with boost test framework. 😄

**Comment by netspiri** [comment]
Initial tests show promise. The latest in-development build works as expected.

There is a minor issue regarding executing template test cases using types within namespaces such as:

```c++
namespace detail
{
    struct MyTag { int m_member; };
} // namespace detail

using TestTypes = boost::mpl::list<detail::MyTag>;

BOOST_AUTO_TEST_CASE_TEMPLATE(TemplateTypeTestExample, T, TestTypes)
{
     BOOST_CHECK_EQUAL(sizeof(T), 4u);
}
```

In such cases, the adapter fails to execute the test. Trying out directly from the command-line had similar effect but I would rather test this further to find the exact root cause (might be a side-effect of #165).

TL;DR; It works for most of the general use cases apart from 1 specific use case.

**Comment by guwirth** [comment]
@raffienficiaud maybe you have an idea?

**Comment by raffienficiaud** [comment]
@guwirth no clue, I am using now proper demangling of the symbol, so in this case you should see `detail__MyTag` if you list the tests

**Comment by netspiri** [comment]
Hi @raffienficiaud and many thanks for the prompt feedback.

When I call `--list_content` using a test module compiled using [Boost 1.63 Beta 1](https://sourceforge.net/projects/boost/files/boost/1.63.0.beta.1/) I get the following (excerpt):
```bat
DataTypeTests*
    TypeTest<int>*
    TypeTest<long>*
    TypeTest<double>*
    TypeTest<unsigned_char>*
    TypeTest<unsigned_char>*
    TypeTest<struct_detail::Custom>*
```
The `--list_content=DOT` output (excerpt) is as follows:
```dot
tu9[shape=Mrecord,fontname=Helvetica,color=green,label="DataTypeTests|c:\users\netspiri\documents\projects\boost unit test adapter\sample\boostunittest\misc\templatemacro.cpp(18)"];
tu1 -> tu9;
{
tu65586[shape=Mrecord,fontname=Helvetica,color=green,label="TypeTest<int>|c:\users\netspiri\documents\projects\boost unit test adapter\sample\boostunittest\misc\templatemacro.cpp(22)"];
tu9 -> tu65586;
tu65587[shape=Mrecord,fontname=Helvetica,color=green,label="TypeTest<long>|c:\users\netspiri\documents\projects\boost unit test adapter\sample\boostunittest\misc\templatemacro.cpp(22)"];
tu9 -> tu65587;
tu65588[shape=Mrecord,fontname=Helvetica,color=green,label="TypeTest<double>|c:\users\netspiri\documents\projects\boost unit test adapter\sample\boostunittest\misc\templatemacro.cpp(22)"];
tu9 -> tu65588;
tu65589[shape=Mrecord,fontname=Helvetica,color=green,label="TypeTest<unsigned_char>|c:\users\netspiri\documents\projects\boost unit test adapter\sample\boostunittest\misc\templatemacro.cpp(22)"];
tu9 -> tu65589;
tu65590[shape=Mrecord,fontname=Helvetica,color=green,label="TypeTest<unsigned_char>|c:\users\netspiri\documents\projects\boost unit test adapter\sample\boostunittest\misc\templatemacro.cpp(22)"];
tu9 -> tu65590;
tu65591[shape=Mrecord,fontname=Helvetica,color=green,label="TypeTest<struct_detail::Custom>|c:\users\netspiri\documents\projects\boost unit test adapter\sample\boostunittest\misc\templatemacro.cpp(22)"];
tu9 -> tu65591;
}
```
I do not know if this is a Windows only quirk or maybe I have linked my sample project incorrectly but this is as far as I can proceed at this point.

**Comment by raffienficiaud** [comment]
I believe those changes are not in the beta1, they were committed on master at rev 4ba85b603cfd257e386144d446428c824a1c6b2a, the 23.11.2016 (changelog ``# [ticket 12531] `--run_test` in Boost 1.62 does not accept test names which contain ':'``), while beta1 dates back to the 9th of November. It should definitely be in 1.63. Sorry for the noise.

Do you have the chance to test with the current master of boost.test ?

**Comment by raffienficiaud** [comment]
Ref for the ticket: https://svn.boost.org/trac/boost/ticket/12531

**Comment by netspiri** [comment]
Tested with master branch and the test passes.

`--list_content` output is now:
```bat
    TypeTest<struct_detail__Custom>*
```
`--list_content=DOT` output is now:
```dot
tu65591[shape=Mrecord,fontname=Helvetica,color=green,label="TypeTest<struct_detail__Custom>|c:\users\netspiri\documents\projects\boost unit test adapter\sample\boostunittest\misc\templatemacro.cpp(22)"];
```

Many thanks for your work and support @raffienficiaud.

P.S. @raffienficiaud Is the `--version` command-line option going to be available in Boost 1.63? If so, that would be a great thing for us since this will be of a huge advantage to us to cater for different Boost.Test versions.

**Comment by raffienficiaud** [comment]
@guwirth thank you for testing so quickly (and with such a good news). Indeed, the `--version` is there for 1.63 (not on the beta1 though), see issue `12530`

**Comment by guwirth** [comment]
@netspiri if everything is working with final boost 1.63 we should release a new version of BUTA.

**Comment by guwirth** [comment]
@netspiri current state with boost 1.63 not clear for me. Everything working with boost 1.63?

**Comment by netspiri** [comment]
Minor issue with Boost 1.63 Beta 1. @raffienficiaud is stating that by the official release of Boost 1.63, this minor issue will be solved.

Given that the issue is relatively minor in my opinion, I would suggest releasing, BUTA 1.0.8 has been a long time coming... User testing confirmation and an update to the version history documentation is all that remains as far as I know.

**Comment by guwirth** [comment]
@netspiri if there is some time left this year finalize version history and release notes please.

**Comment by gastart** [comment]
Hello @guwirth. Since I know that @netspiri is on holiday for another week I thought of generating the release notes myself.

I've updated the release notes for both 1.0.8.0 and 1.0.7.0 (because it was missing).

I used a different git tool, and by mistake I forgot to branch and ended up committing directly to your master (73d54cf). I apologize for this.  Do you want me to revert and create a pull request?

**Comment by guwirth** [comment]
@netspiri & @gastart final boost 1.63 is available. Can you do a final test with released version please.

**Comment by netspiri** [comment]
@guwirth Latest development build seems to works OK with the official release of Boost 1.63

**Comment by guwirth** [comment]
@netspiri ready to release?

**Comment by netspiri** [comment]
@guwirth Can we squeeze in one final refactoring exercise before release? Other than that, there are no expected major changes.
